src/components/VeggyBody/VeggyBody.jsx

```diff
@@ -0,0 +1,83 @@
+import './VeggyBody.css'
+import VeggyProductItem from "./VeggyProductItem/VeggyProductItem.jsx";
+import {useState} from "react";
+import ModalProductItemImage from "./Modal/Modal.jsx";
+
+export default function VeggyBody(props) {
+    const [modalActive, setModalActive] = useState(false);
+
+
+    const [imageValue, setImageValue] = useState({src: null, name: null, price: null});
+    function zoomProductImage (e) {
+        setModalActive(!modalActive);
```

Comment:
Май за звичку правило: якщо ти обновляєш стейт базуючись на його попередньому значенні, - то обновляй його використовуючи колбек:
```suggestion
        setModalActive(modalActive => !modalActive);
```